### ISSUE TITLE
I cant sign into my account

### ISSUE THREAD

**Issue by NickiWillson** [DSJAS/DSJAS#33]
Hi. I finished the installation and the last screen asked to setup my username and password. I entered them, but when i then try to login it says 'invalid username the entered username was not found' do you have anyway to resolve this if possible? The site looks great by the way!

**Comment by ethanv2** [comment]
Hi there. Please make sure you are signing in to the admin panel and not the bank login. DSJAS has a login for the admin panel and for the fake bank dashboard. Please see [here](https://github.com/DSJAS/DSJAS/blob/master/docs/administration/Access%20management.md) for help on logging in to the admin panel.

As for the fake bank, you get a default user called "edna" for the bank. For some specific instructions, see [this section](https://github.com/DSJAS/DSJAS/blob/master/docs/install/First%20steps%20in%20the%20program.md#go-to-the-homepage-and-survey-the-sights) of the user manual.

For future reference, any help you need about the site can usually be found in the [docs directory](https://github.com/DSJAS/DSJAS/tree/master/docs) in the site's source.

I'm glad you like the site - if you need any more help, just send me back a reply and I'll get back to you as soon as I can. Good luck and happy scambaiting!

**Comment by ethanv2** [comment]
Hi there. Please make sure you are signing in to the admin panel and not the bank login. DSJAS has a login for the admin panel and for the fake bank dashboard. Please see [here](https://github.com/DSJAS/DSJAS/blob/master/docs/administration/Access%20management.md) for help on logging in to the admin panel.

As for the fake bank, you get a default user called "edna" for the bank. For some specific instructions, see [this section](https://github.com/DSJAS/DSJAS/blob/master/docs/install/First%20steps%20in%20the%20program.md#go-to-the-homepage-and-survey-the-sights) of the user manual.

For future reference, any help you need about the site can usually be found in the [docs directory](https://github.com/DSJAS/DSJAS/tree/master/docs) in the site's source.

I'm glad you like the site - if you need any more help, just send me back a reply and I'll get back to you as soon as I can. Good luck and happy scambaiting!

**Comment by NickiWillson** [comment]
Ah ok thats fine. Im actually having another issue. I have set up the URL spoofing as per the docs and when i enter the URL I have put into the hosts file, i get a 403 forbidden page with this message: 
"Forbidden

You don't have permission to access this resource.
Apache/2.4.48 (Win64) OpenSSL/1.1.1k PHP/8.0.9 Server at brownings.bank.net Port 443"

I noticed the server is 8.0.9 and not 127.0.0.1. Im not sure if that is the issue or how i would resolve that.

**Comment by ethanv2** [comment]
Can you post a copy/link to your Apache config and error logs? They are usually located either in the same directory as your Apache executable, or in locations like ``/etc/httpd.conf`` or ``/var/log/apache2``.

The fact that this still works on localhost but nowhere else suggests that your Apache is configured to only allow connections from the localhost domain, which is easily fixable once we pinpoint the problem.

In the error log, we'll be looking for error lines that look like this:

```
access to /blah denied (filesystem path
'/home/whoever/Sites/mySite') because xyz
```

**Comment by NickiWillson** [comment]
[Sun Aug 15 11:39:06.733566 2021] [ssl:warn] [pid 12400:tid 580] AH01909: www.example.com:443:0 server certificate does NOT include an ID which matches the server name
[Sun Aug 15 11:39:06.764821 2021] [ssl:warn] [pid 12400:tid 580] AH01909: www.example.com:443:0 server certificate does NOT include an ID which matches the server name
[Sun Aug 15 11:39:06.796072 2021] [mpm_winnt:notice] [pid 12400:tid 580] AH00455: Apache/2.4.48 (Win64) OpenSSL/1.1.1k PHP/8.0.9 configured -- resuming normal operations
[Sun Aug 15 11:39:06.796072 2021] [mpm_winnt:notice] [pid 12400:tid 580] AH00456: Apache Lounge VS16 Server built: May 18 2021 10:45:56
[Sun Aug 15 11:39:06.796072 2021] [core:notice] [pid 12400:tid 580] AH00094: Command line: 'C:\\xampp\\apache\\bin\\httpd.exe -d C:/xampp/apache'
[Sun Aug 15 11:39:06.808578 2021] [mpm_winnt:notice] [pid 12400:tid 580] AH00418: Parent: Created child process 3448
[Sun Aug 15 11:39:07.159408 2021] [ssl:warn] [pid 3448:tid 644] AH01909: www.example.com:443:0 server certificate does NOT include an ID which matches the server name
[Sun Aug 15 11:39:07.191521 2021] [ssl:warn] [pid 3448:tid 644] AH01909: www.example.com:443:0 server certificate does NOT include an ID which matches the server name
[Sun Aug 15 11:39:07.211583 2021] [mpm_winnt:notice] [pid 3448:tid 644] AH00354: Child: Starting 150 worker threads.
[Sun Aug 15 11:39:32.380672 2021] [ssl:warn] [pid 7816:tid 716] AH01909: www.example.com:443:0 server certificate does NOT include an ID which matches the server name
[Sun Aug 15 11:39:32.418899 2021] [ssl:warn] [pid 7816:tid 716] AH01909: www.example.com:443:0 server certificate does NOT include an ID which matches the server name
[Sun Aug 15 11:39:32.434528 2021] [mpm_winnt:notice] [pid 7816:tid 716] AH00455: Apache/2.4.48 (Win64) OpenSSL/1.1.1k PHP/8.0.9 configured -- resuming normal operations
[Sun Aug 15 11:39:32.434528 2021] [mpm_winnt:notice] [pid 7816:tid 716] AH00456: Apache Lounge VS16 Server built: May 18 2021 10:45:56
[Sun Aug 15 11:39:32.434528 2021] [core:notice] [pid 7816:tid 716] AH00094: Command line: 'c:\\xampp\\apache\\bin\\httpd.exe -d C:/xampp/apache'
[Sun Aug 15 11:39:32.434528 2021] [mpm_winnt:notice] [pid 7816:tid 716] AH00418: Parent: Created child process 9776
[Sun Aug 15 11:39:32.770023 2021] [ssl:warn] [pid 9776:tid 672] AH01909: www.example.com:443:0 server certificate does NOT include an ID which matches the server name
[Sun Aug 15 11:39:32.801279 2021] [ssl:warn] [pid 9776:tid 672] AH01909: www.example.com:443:0 server certificate does NOT include an ID which matches the server name
[Sun Aug 15 11:39:32.816905 2021] [mpm_winnt:notice] [pid 9776:tid 672] AH00354: Child: Starting 150 worker threads.
[Sun Aug 15 11:41:45.991503 2021] [ssl:warn] [pid 3948:tid 512] AH01909: www.example.com:443:0 server certificate does NOT include an ID which matches the server name
[Sun Aug 15 11:41:46.024772 2021] [core:warn] [pid 3948:tid 512] AH00098: pid file C:/xampp/apache/logs/httpd.pid overwritten -- Unclean shutdown of previous Apache run?
[Sun Aug 15 11:41:46.024772 2021] [ssl:warn] [pid 3948:tid 512] AH01909: www.example.com:443:0 server certificate does NOT include an ID which matches the server name
[Sun Aug 15 11:41:46.040400 2021] [mpm_winnt:notice] [pid 3948:tid 512] AH00455: Apache/2.4.48 (Win64) OpenSSL/1.1.1k PHP/8.0.9 configured -- resuming normal operations
[Sun Aug 15 11:41:46.040400 2021] [mpm_winnt:notice] [pid 3948:tid 512] AH00456: Apache Lounge VS16 Server built: May 18 2021 10:45:56
[Sun Aug 15 11:41:46.040400 2021] [core:notice] [pid 3948:tid 512] AH00094: Command line: 'c:\\xampp\\apache\\bin\\httpd.exe -d C:/xampp/apache'
[Sun Aug 15 11:41:46.040400 2021] [mpm_winnt:notice] [pid 3948:tid 512] AH00418: Parent: Created child process 7172
[Sun Aug 15 11:41:46.372035 2021] [ssl:warn] [pid 7172:tid 664] AH01909: www.example.com:443:0 server certificate does NOT include an ID which matches the server name
[Sun Aug 15 11:41:46.418913 2021] [ssl:warn] [pid 7172:tid 664] AH01909: www.example.com:443:0 server certificate does NOT include an ID which matches the server name
[Sun Aug 15 11:41:46.434539 2021] [mpm_winnt:notice] [pid 7172:tid 664] AH00354: Child: Starting 150 worker threads.
[Sun Aug 15 11:41:55.600603 2021] [php:warn] [pid 7172:tid 1860] [client 127.0.0.1:61763] PHP Warning:  Undefined array key "setup_authorised" in D:\\BrowningsBank\\include\\install\\Utils.php on line 104
[Sun Aug 15 11:57:32.451570 2021] [authz_core:error] [pid 7172:tid 1856] [client 127.0.0.1:61983] AH01630: client denied by server configuration: C:/xampp/htdocs/
[Sun Aug 15 11:57:32.852624 2021] [authz_core:error] [pid 7172:tid 1856] [client 127.0.0.1:61983] AH01630: client denied by server configuration: C:/xampp/htdocs/favicon.ico, referer: https://brownings.bank.net/
[Sun Aug 15 12:00:40.477655 2021] [authz_core:error] [pid 7172:tid 1856] [client 127.0.0.1:61992] AH01630: client denied by server configuration: C:/xampp/htdocs/
[Sun Aug 15 12:00:40.646548 2021] [authz_core:error] [pid 7172:tid 1856] [client 127.0.0.1:61992] AH01630: client denied by server configuration: C:/xampp/htdocs/favicon.ico, referer: https://brownings.bank.net/
[Sun Aug 15 12:02:45.216115 2021] [ssl:warn] [pid 9056:tid 528] AH01909: www.example.com:443:0 server certificate does NOT include an ID which matches the server name
[Sun Aug 15 12:02:45.250520 2021] [core:warn] [pid 9056:tid 528] AH00098: pid file C:/xampp/apache/logs/httpd.pid overwritten -- Unclean shutdown of previous Apache run?
[Sun Aug 15 12:02:45.250520 2021] [ssl:warn] [pid 9056:tid 528] AH01909: www.example.com:443:0 server certificate does NOT include an ID which matches the server name
[Sun Aug 15 12:02:45.281774 2021] [mpm_winnt:notice] [pid 9056:tid 528] AH00455: Apache/2.4.48 (Win64) OpenSSL/1.1.1k PHP/8.0.9 configured -- resuming normal operations
[Sun Aug 15 12:02:45.281774 2021] [mpm_winnt:notice] [pid 9056:tid 528] AH00456: Apache Lounge VS16 Server built: May 18 2021 10:45:56
[Sun Aug 15 12:02:45.281774 2021] [core:notice] [pid 9056:tid 528] AH00094: Command line: 'c:\\xampp\\apache\\bin\\httpd.exe -d C:/xampp/apache'
[Sun Aug 15 12:02:45.281774 2021] [mpm_winnt:notice] [pid 9056:tid 528] AH00418: Parent: Created child process 6276
[Sun Aug 15 12:02:45.619871 2021] [ssl:warn] [pid 6276:tid 668] AH01909: www.example.com:443:0 server certificate does NOT include an ID which matches the server name
[Sun Aug 15 12:02:45.651133 2021] [ssl:warn] [pid 6276:tid 668] AH01909: www.example.com:443:0 server certificate does NOT include an ID which matches the server name
[Sun Aug 15 12:02:45.666751 2021] [mpm_winnt:notice] [pid 6276:tid 668] AH00354: Child: Starting 150 worker threads.
[Sun Aug 15 12:04:57.036318 2021] [authz_core:error] [pid 6276:tid 1860] [client 127.0.0.1:62023] AH01630: client denied by server configuration: C:/xampp/htdocs/
[Sun Aug 15 12:04:57.152252 2021] [authz_core:error] [pid 6276:tid 1860] [client 127.0.0.1:62023] AH01630: client denied by server configuration: C:/xampp/htdocs/favicon.ico, referer: https://brownings.bank.net/
[Sun Aug 15 12:09:19.619922 2021] [authz_core:error] [pid 6276:tid 1860] [client 127.0.0.1:62049] AH01630: client denied by server configuration: C:/xampp/htdocs/
[Sun Aug 15 12:09:19.735757 2021] [authz_core:error] [pid 6276:tid 1860] [client 127.0.0.1:62049] AH01630: client denied by server configuration: C:/xampp/htdocs/favicon.ico, referer: https://brownings.bank.net/
[Sun Aug 15 12:18:54.775300 2021] [php:error] [pid 6276:tid 1860] [client 127.0.0.1:62148] PHP Fatal error:  Uncaught ArgumentCountError: The number of variables must match the number of parameters in the prepared statement in D:\\BrowningsBank\\include\\DB.php:254\nStack trace:\n#0 D:\\BrowningsBank\\include\\DB.php(254): mysqli_stmt->bind_param('i', '2')\n#1 D:\\BrowningsBank\\include\\Banking.php(395): DB->prepareQuery(Object(PreparedStatement))\n#2 D:\\BrowningsBank\\admin\\bank\\accounts.php(35): closeAccount('2')\n#3 {main}\n  thrown in D:\\BrowningsBank\\include\\DB.php on line 254, referer: http://localhost/admin/bank/accounts.php?closeAccount=2&csrf=xYDT16yrLpLcXGpQy5fzKFjQXcleOonP
[Sun Aug 15 12:19:19.116384 2021] [php:error] [pid 6276:tid 1860] [client 127.0.0.1:62166] PHP Fatal error:  Uncaught ArgumentCountError: The number of variables must match the number of parameters in the prepared statement in D:\\BrowningsBank\\include\\DB.php:254\nStack trace:\n#0 D:\\BrowningsBank\\include\\DB.php(254): mysqli_stmt->bind_param('i', '4')\n#1 D:\\BrowningsBank\\include\\Banking.php(395): DB->prepareQuery(Object(PreparedStatement))\n#2 D:\\BrowningsBank\\admin\\bank\\accounts.php(35): closeAccount('4')\n#3 {main}\n  thrown in D:\\BrowningsBank\\include\\DB.php on line 254, referer: http://localhost/admin/bank/accounts.php?closeAccount=4&csrf=B4OnyRXDQVcurXzmzV4OEKKjy9gFnFoP
[Sun Aug 15 12:21:07.301906 2021] [php:error] [pid 6276:tid 1860] [client 127.0.0.1:62249] PHP Fatal error:  Uncaught ArgumentCountError: The number of variables must match the number of parameters in the prepared statement in D:\\BrowningsBank\\include\\DB.php:254\nStack trace:\n#0 D:\\BrowningsBank\\include\\DB.php(254): mysqli_stmt->bind_param('i', '1')\n#1 D:\\BrowningsBank\\include\\Banking.php(395): DB->prepareQuery(Object(PreparedStatement))\n#2 D:\\BrowningsBank\\admin\\bank\\accounts.php(35): closeAccount('1')\n#3 {main}\n  thrown in D:\\BrowningsBank\\include\\DB.php on line 254, referer: http://localhost/admin/bank/accounts.php?closeAccount=1&csrf=Ki5g6ESdIOS9uqMOgryYRNTO2ivhFv3W

**Comment by NickiWillson** [comment]
Im not sure if that is the correct error log

**Comment by NickiWillson** [comment]
[error.log](https://github.com/DSJAS/DSJAS/files/6988125/error.log) sorry I didnt realise I could drag and drop files

**Comment by ethanv2** [comment]
The offending lines are the following:

```
[Sun Aug 15 11:57:32.451570 2021] [authz_core:error] [pid 7172:tid 1856] [client 127.0.0.1:61983] AH01630: client denied by server configuration: C:/xampp/htdocs/
[Sun Aug 15 11:57:32.852624 2021] [authz_core:error] [pid 7172:tid 1856] [client 127.0.0.1:61983] AH01630: client denied by server configuration: C:/xampp/htdocs/favicon.ico, referer: https://brownings.bank.net/
[Sun Aug 15 12:00:40.477655 2021] [authz_core:error] [pid 7172:tid 1856] [client 127.0.0.1:61992] AH01630: client denied by server configuration: C:/xampp/htdocs/
[Sun Aug 15 12:00:40.646548 2021] [authz_core:error] [pid 7172:tid 1856] [client 127.0.0.1:61992] AH01630: client denied by server configuration: C:/xampp/htdocs/favicon.ico, referer: https://brownings.bank.net/
```

Somewhere in your configuration file, you are denying users who do not come from localhost. All we need to find now is the place where. Would you mind attaching your server config like you did the error log, so that I can find the problem?

**Comment by NickiWillson** [comment]
would that be httpd.conf? If so i have tried attaching that file and the file format is not supported. Could i copy the file and change it to .txt or .log?

**Comment by NickiWillson** [comment]
[httpd.log](https://github.com/DSJAS/DSJAS/files/6988162/httpd.log) Just incase that is the correct file

**Comment by ethanv2** [comment]
Does the custom domain when accessed using just HTTP, rather than HTTPS?

**Comment by NickiWillson** [comment]
If i type http:// it just redirects to https://

**Comment by ethanv2** [comment]
Try changing the text after "ServerName" in your config file from "localhost:80" to the name you want to use. Be sure to restart your server using the Xampp control panel!

**Comment by NickiWillson** [comment]
Just for reference, this is what my hosts file displays now (just to make sure i havent messed anything up there):
# Copyright (c) 1993-2009 Microsoft Corp.
#
# This is a sample HOSTS file used by Microsoft TCP/IP for Windows.
#
# This file contains the mappings of IP addresses to host names. Each
# entry should be kept on an individual line. The IP address should
# be placed in the first column followed by the corresponding host name.
# The IP address and the host name should be separated by at least one
# space.
#
# Additionally, comments (such as these) may be inserted on individual
# lines or following the machine name denoted by a '#' symbol.
#
# For example:
#
#      102.54.94.97     rhino.acme.com          # source server
#       38.25.63.10     x.acme.com              # x client host

# localhost name resolution is handled within DNS itself.
#	127.0.0.1       localhost
#	::1             localhost

# DSJAS Domain Mapping
127.0.0.1   brownings.bank.net

**Comment by NickiWillson** [comment]
I have changed the text after "servername" and still the same issue

**Comment by ethanv2** [comment]
Ok, after some quick research, this may have been caused by some default XAMPP HTTPS settings. If you open the file "C:\xampp\apache\conf\extra\httpd-ssl.conf", you will see a line around line 86 which reads ``DocumentRoot "/opt/lampp/htdocs"``. You need to change this to your actual document root, which I believe is ``D:\BrowningsBank``.

Try this and get back to me. If this fixes the issue, I will amend the documentation to tell other users who are using HTTPS of this issue.

**Comment by NickiWillson** [comment]
Ok im really sorry, i have found the issue! In the installation steps, where you are told to select config in XAMPP and change the document root to the path we installed the zip file in.......in XAMPP there are 3 apache config files, 2 of them contain a "document root" setting and 1 does not. I only changed it in the first config file and not in the second. I have just changed it in the second and it has worked! Im really sorry for wasting your time. And again thankyou for the website it is amazing!
